src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -275,22 +322,85 @@ def _getNumberOfTiles(self):
                      for s, f in zip(self._area_size, reliable_fov)]
         nx = math.ceil(area_size[0] / reliable_fov[0])
         ny = math.ceil(area_size[1] / reliable_fov[1])
-        logging.debug("Calculated number of tiles nx= %s, ny= %s" % (nx, ny))
 
         # We have a little bit more tiles than needed, we then have two choices
         # on how to spread them:
         # 1. Increase the total area acquired (and keep the overlap)
         # 2. Increase the overlap (and keep the total area)
         # We pick alternative 1 (no real reason)
-        center = (self._area[0] + self._area[2]) / 2, (self._area[1] + self._area[3]) / 2
+        xmin, ymin, xmax, ymax = self._polygon.bounds
+        center = ((xmin + xmax) / 2, (ymin + ymax) / 2)
         total_size = nx * reliable_fov[0], ny * reliable_fov[1]
-
-        # Compute the top-left of the "bigger" area, and from it, shift by half
-        # the size of the smallest tile.
-        starting_pos = {'x': center[0] - total_size[0] / 2 + reliable_fov[0] / 2,  # left
-                        'y': center[1] + total_size[1] / 2 - reliable_fov[1] / 2}  # top
-
-        return (nx, ny), starting_pos
+        xmin = center[0] - total_size[0] / 2
+        ymax = center[1] + total_size[1] / 2
+
+        # Create an empty grid for storing intersected tiles
+        tile_grid = numpy.zeros((ny, nx), dtype=bool)
+
+        def get_possible_intersected_tiles(row_pairs, col_pairs):
+            """Bresenham's line algorithm to determine the possible intersected tiles."""
+            tiles = set()
+            for (row1, row2), (col1, col2) in zip(row_pairs, col_pairs):
+                dx = abs(row2 - row1)
+                dy = abs(col2 - col1)
+                sx = 1 if row1 < row2 else -1
+                sy = 1 if col1 < col2 else -1
+                err = dx - dy
+
+                while True:
+                    tiles.add((row1, col1))
+                    # Add adjacent neighbors to ensure complete coverage
+                    tiles.add((row1 + sx, col1))
+                    tiles.add((row1 - sx, col1))
+                    tiles.add((row1, col1 + sy))
+                    tiles.add((row1, col1 - sy))
+                    if row1 == row2 and col1 == col2:
+                        break
+                    e2 = err * 2
+                    if e2 > -dy:
+                        err -= dy
+                        row1 += sx
+                    if e2 < dx:
+                        err += dx
+                        col1 += sy
+            return tiles
+
+        # Vectorized conversion of polygon points to grid coordinates
```

Comment:
I think this whole algorithm that computes which tile needs to be scanned should go into a dedicated function. That would make the intention much clearer (than in a function supposed to "just" count the number of tiles).

Reply:
Renamed the function, and later find the number of tiles by using `len`